sql/src/main/java/io/crate/expression/reference/file/LineContext.java

```diff
@@ -23,32 +23,40 @@
 
 import io.crate.metadata.ColumnIdent;
 import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.ElasticsearchParseException;
 import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.compress.NotXContentException;
+import org.elasticsearch.common.lucene.BytesRefs;
 import org.elasticsearch.common.xcontent.XContentHelper;
 import org.elasticsearch.common.xcontent.XContentType;
 
 import javax.annotation.Nullable;
+import java.net.URI;
 import java.util.Map;
 
 public class LineContext {
 
     private byte[] rawSource;
     private Map<String, Object> parsedSource;
+    private BytesRef currentUri;
+    private String currentUriFailure;
 
     @Nullable
-    public BytesRef sourceAsBytesRef() {
+    BytesRef sourceAsBytesRef() {
         if (rawSource != null) {
             return new BytesRef(rawSource);
         }
         return null;
     }
 
-    public Map<String, Object> sourceAsMap() {
+    Map<String, Object> sourceAsMap() {
         if (parsedSource == null) {
             try {
                 parsedSource = XContentHelper.convertToMap(new BytesArray(rawSource), false, XContentType.JSON).v2();
             } catch (NullPointerException e) {
                 return null;
+            } catch (ElasticsearchParseException | NotXContentException e) {
+                throw new RuntimeException("JSON parser error: " + e.getMessage(), e);
```

Comment:
why do we swallow NullPointerExceptions here but not these two other exceptions?

Reply:
hm right, we should bubble the NPE as well. the others are just catched to transform them in a more meaningful exception/message.